Assets/MixedRealityToolkit/Utilities/Editor/Preferences/ProjectPreferences.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    /// <summary>
+    /// Utility to save preferences that should be saved per project (i.e to source control) across MRTK. Supports primitive preferences bool, int, and float
+    /// </summary>
+    public class ProjectPreferences : ScriptableObject
+    {
+        // Dictionary is not Serializable by default and furthermore System.object is not Serializable
+        // Thus, it is difficult to create a generic data bag. Instead we will create instances for each key preference types
```

Comment:
Is there a reason we went down the route of "generic data bag", instead of creating a specific "MRTKPreferences" `ScriptableObject` (like https://github.com/microsoft/MixedRealityToolkit-Unity/pull/6309 started to do and I have in some form in a local branch)?

My thought is that we know exactly what our possible values are going to be and which preferences are going to need to be stored, so it seems like a lot to fully recreate the existing `EditorPrefs` as a generic thing as opposed to specifically coding the settings / preferences. Might make the `SerializedObject` (and its inspector) a little more readable and customizable, too, since we specifically know what data it'll store.

Reply:
There are also some docs around using `ScriptableObject`s as `SettingsProvider`s: https://docs.unity3d.com/ScriptReference/SettingsProvider.html (which might still be possible here? but probably easier when we know the contents)

Reply:
The impetus of using a data bag was to be more flexible and extensible. Answering your initial questions:

1) It actually isn't recreation because the feature does not exist yet. EditorPreferences/EditorPref do not work as project settings. 
2) #6309 had to create their own SO and the work of loading/identifying a file. It's actually more files and more coding for everyone to manage their own instance

I'd say MixedRealityPreferences.cs is the class of known tracked properties. 

We could create a single SO object with one loading utility that contains all Project Preferences explicitly (i.e one property line for each thing you need) and every new consumer has to go in and add their fields. Basically it would be a one off header file with every definition.

The problem with that is 
1) People will keep adding to it and eventually it will become disorganized and possibly outdated etc. (Less important)
2) Really only MRTK-internal can utilize the class (more important)

Regarding inspectors, selecting the SO file and viewing it in inspector is not golden path or expected path at all. We need to consider it and handle it as a possibility but the file should remain mostly hidden. 

I've looked at the SettingsProvider docs and MixedRealityPreference.cs is using it. What is the item of value in there though that you are referencing? It has some example code but I don't see any callouts for best practices or warnings etc. 

The key is I don't expect all Project Preferences to always fit nicely into that Project Settings > Preferences view. An example of that is the build settings. 

Thoughts?

Reply:
>The impetus of using a data bag was to be more flexible and extensible.

I like a data bag for this reason. As the toolkit grows there'll be breaking config changes, it's inevitable. Tools like FormallySerializedAs can only do so much. With a data bag we can filter stuff on the way in / out and detect outdated keys, migrate settings, etc. etc.